src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -308,6 +313,21 @@ def _slice(
         assert common.is_relative_index_sequence(slice_)
         return new_domain, slice_
 
+    def _refactor_slice(
+        self, idx: slice
+    ) -> tuple[common.Dimension, common.UnitRange] | common.AnyIndexSpec:
+        if isinstance(idx.start, tuple) or isinstance(idx.stop, tuple):
+            is_start_none: bool = idx.start is None
```

Comment:
I believe introducing this local variable is not good style. The name is roughly as long as the rhs expression and harder to read, because of the indirection.